contracts/ConfidentialERC20/ConfidentialERC20.sol

```diff
@@ -232,6 +254,25 @@ abstract contract ConfidentialERC20 is Ownable, IConfidentialERC20, IERC20Metada
         _totalSupply += value;
     }
 
+    event BurnRequested(uint256 requestId, address account, uint64 amount);
+
+    event BurnRequestCancelled(uint256 requestID);
+
+    error BurnRequestDoesNotExist(uint256 requestId);
+
+    function _cancelBurn(uint256 requestId) internal virtual {
```

Comment:
This could be exposed to the original sender/unwrapper if they wanted to cancel in-flight. We ought to be able to rely on the callback being called though, so we shouldn't need this _just_ for cleaning up in-flight burns.


The EOA cna use the `BurnRequested` event to keep track of this requestId so it's possible to give them cancellation